snmp/datadog_checks/snmp/snmp.py

```diff
@@ -301,8 +301,9 @@ def _start_discovery(self):
                     write_persistent_cache(self.check_id, json.dumps([]))
                     break
                 instance = copy.deepcopy(self.instance)
-                instance.pop('network_address')
+                network_address = instance.pop('network_address')
                 instance['ip_address'] = host
+                instance['autodiscovery_subnet'] = network_address
```

Comment:
This seems hacky (`autodiscovery_subnet` is not a public instance option), is there a way we can add the tags directly? Maybe editing `host_config.tags` manually below?

Probably similar suggestion to Thomas' https://github.com/DataDog/integrations-core/pull/6941/files#r444748253

Then maybe this makes #5790 irrelevant?

Reply:
@florimondmanca Updated to use tags instead.

Not sure how this related to https://github.com/DataDog/integrations-core/pull/5790 ?

Reply:
I modified `instance['tags']` instead of changing `host_config.tags` to better mimic the behaviour of `auto_conf.yaml`.